modules/main/src/main/java/module-info.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+module org.elasticsearch.rest.main {
```

Comment:
The name "main" is too generic IMO (it would be reasonable for someone looking under the `modules/` dir to think this was the main entrypoint for the Elasticsearch process). I understand the handler has always been called "main", but as we are moving it to a module I think we should take the opportunity to improve the name. My suggestion, at least for the directory, is "root-endpoint". Perhaps the module could be "org.elasticsearch.rest.root".